larpandoracontent/LArHelpers/LArHitWidthHelper.cc

```diff
@@ -0,0 +1,292 @@
+
+    if (clusterToParametersMap.empty())
+        throw StatusCodeException(STATUS_CODE_NOT_INITIALIZED);
+
+    const auto clusterParametersIter(clusterToParametersMap.find(pCluster));
+
+    if (clusterParametersIter == clusterToParametersMap.end())
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    return clusterParametersIter->second;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHitWidthHelper::ConstituentHitVector LArHitWidthHelper::GetConstituentHits(const Cluster *const pCluster, const float maxConstituentHitWidth,
+        const float hitWidthScalingFactor, const bool isUniform)
+{
+    const OrderedCaloHitList &orderedCaloHitList(pCluster->GetOrderedCaloHitList());
+
+    if (orderedCaloHitList.empty())
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    ConstituentHitVector constituentHitVector;
+    for (const OrderedCaloHitList::value_type &mapEntry : orderedCaloHitList)
+    {
+        for (const CaloHit *const pCaloHit : *mapEntry.second) 
+        {
+            const float hitWidth = pCaloHit->GetCellSize1() * hitWidthScalingFactor;
+            const unsigned int numberOfConstituentHits = ceil(hitWidth / maxConstituentHitWidth);
+            if (isUniform)
+            {
+                SplitHitIntoConstituents(pCaloHit, pCluster, numberOfConstituentHits, maxConstituentHitWidth, constituentHitVector);
+            }
+            else
+            {
+                const float constituentHitWidth = hitWidth / numberOfConstituentHits;
+                SplitHitIntoConstituents(pCaloHit, pCluster, numberOfConstituentHits, constituentHitWidth, constituentHitVector);
+            }
+        }
+    }
+    
+    return constituentHitVector;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArHitWidthHelper::SplitHitIntoConstituents(const CaloHit *const pCaloHit, const Cluster *const pCluster, const unsigned int numberOfConstituentHits, const float constituentHitWidth,
+        LArHitWidthHelper::ConstituentHitVector &constituentHitVector)
+{
+    // begin at the central hit position
+    CartesianVector positionAlongHit(pCaloHit->GetPositionVector());
+    if (numberOfConstituentHits % 2 == 1)
+    {
+        // if odd number of constituents, include the central position
+        constituentHitVector.push_back(ConstituentHit(positionAlongHit, constituentHitWidth, pCluster));
+
+        // find remaining positions either side of central position
+	    for (unsigned int i = 0; i < ceil((numberOfConstituentHits - 1) / 2); ++i)
+	    {
+            positionAlongHit += CartesianVector(constituentHitWidth, 0.f, 0.f);
+            constituentHitVector.push_back(ConstituentHit(positionAlongHit, constituentHitWidth, pCluster));
```

Comment:
Indentation is off here

Reply:
sorry! my laptop seems to be changing the indents again so i'll try to remember checking things in nano first